stomp/src/main/scala/akka/stream/alpakka/stomp/client/SinkStage.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.stomp.client
+
+import akka.Done
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler}
+import akka.stream.{ActorAttributes, Attributes, Inlet, SinkShape}
+import io.vertx.ext.stomp.{Frame, StompClientConnection}
+
+import scala.concurrent.{Future, Promise}
+
+/**
+ * Connects to a STOMP server upon materialization and sends incoming messages to the server.
+ * Each materialized sink will create one connection to the server.
+ */
+final class SinkStage(settings: ConnectorSettings)
```

Comment:
At a min, we should have a Source and Flow. Sink is a convenience on Flow. Please see the guidelines for more info.

Reply:
Flows should also permit an opaque element to be carried through them. I think this is also in the guidelines.

Reply:
@huntc we do have a Source in SourceStage.scala. 
Regarding the Flow, at this stage, it's not clear to me what do you mean. Could you point me the place in the guidelines that talk about it, since I'm unable to find it. Furthermore, and an example of a flow with the opaque element. Neither I was able to find it in the guidelines. Thanks

Reply:
Here are the guidelines I referenced: https://github.com/akka/alpakka/blob/master/contributor-advice.md

Reply:
Yes, I had read those. I can't find what you mention. Nevertheless, the basic blocks to construct anything, AFAIU is a Source and a Sink. Later we could add flows, that connects those sources and sink, depending in the most common use cases. 

Anyways, what kind of Flow are you imaging? A flow in a stream, that has a side effect of sending the whatever is being passed to a stomp server? That is what you meant by opaque element?

Reply:
Hopefully, these links will help:

https://github.com/akka/alpakka/blob/master/contributor-advice.md#public-factory-methods
https://github.com/akka/alpakka/blob/master/contributor-advice.md#flows

You generally construct sinks from flows:

https://github.com/akka/alpakka/blob/master/contributor-advice.md#keep-the-code-dry

Some examples, also illustrating the passing through of an element:

```
  /**
    * Append a command request to a queue
    * @tparam A The type of data to carry through
    * @return a flow that takes a command and returns with an acknowledgement
    */
  def flow[A]: Flow[CommandRequest[A], CommandReply[A], NotUsed]

  /**
    * Append a command request to a queue via a sink for convenience
    * @tparam A The type of data to carry through
    * @return a sink that takes a command and returns with an acknowledgement
    */
  def sink[A]: Sink[CommandRequest[A], Future[Done]] =
    flow.toMat(Sink.ignore)(Keep.right)
```

Reply:
Oh, I see what you say. For me, the example is key. I had already read the guidelines links a month ago and didn't get this idea. The example is crystal, can it be added to the contributor-advice? (is there any copyright restriction on the example you wrote?)

Reply:
I also see the approach you propose is cleaner and more powerful. I would love to have hear this when **I outlined the shape and plan for this contribution on feb 26th** ! A month ago! (https://github.com/akka/alpakka/issues/514#issuecomment-368509908) 

For the moment I'm reluctant to switch approaches. Since this works for several use cases that I find useful.